lib/ansible/parsing/vault/__init__.py

```diff
@@ -853,6 +853,7 @@ def _edit_file_helper(self, filename, secret,
         except:
```

Comment:
actually, this should be a 'finally' instead (also remove 848)

Reply:
I tried it with finally before but that seems to be wrong. If its finally, the tmp_path is removed after try-catch and the read_data call right after it fails.

Reply:
we just might need to change the scope of where the 'finally' is set

Reply:
pushed more stuff to try block and added finally clause. Seems better.
Also returned the mock object and fixed it to use "touch" as editor. Test seems to pass now but I think there should be still tests to see that VISUAL and EDITOR env variables are properly handled.